// @clerk/nextjs/dist/cjs/server/keyless-node.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var keyless_node_exports = {};
__export(keyless_node_exports, {
  keyless: () => keyless
});
module.exports = __toCommonJS(keyless_node_exports);
var import_keyless = require("@clerk/shared/keyless");
var import_createClerkClient = require("./createClerkClient");
var import_utils = require("./fs/utils");
function createFileStorage() {
  const fs = (0, import_utils.nodeFsOrThrow)();
  const path = (0, import_utils.nodePathOrThrow)();
  const cwd = (0, import_utils.nodeCwdOrThrow)();
  return (0, import_keyless.createNodeFileStorage)(fs, path, {
    cwd,
    frameworkPackageName: "@clerk/nextjs"
  });
}
let keylessServiceInstance = null;
function keyless() {
  if (!keylessServiceInstance) {
    const client = (0, import_createClerkClient.createClerkClientWithOptions)({});
    keylessServiceInstance = (0, import_keyless.createKeylessService)({
      storage: createFileStorage(),
      api: {
        async createAccountlessApplication(requestHeaders) {
          try {
            return await client.__experimental_accountlessApplications.createAccountlessApplication({
              requestHeaders
            });
          } catch {
            return null;
          }
        },
        async completeOnboarding(requestHeaders) {
          try {
            return await client.__experimental_accountlessApplications.completeAccountlessApplicationOnboarding({
              requestHeaders
            });
          } catch {
            return null;
          }
        }
      },
      framework: "nextjs"
    });
  }
  return keylessServiceInstance;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  keyless
});
