// @clerk/nextjs/dist/cjs/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var index_exports = {};
__export(index_exports, {
  APIKeys: () => import_uiComponents.APIKeys,
  AuthenticateWithRedirectCallback: () => import_controlComponents.AuthenticateWithRedirectCallback,
  ClerkDegraded: () => import_controlComponents.ClerkDegraded,
  ClerkFailed: () => import_controlComponents.ClerkFailed,
  ClerkLoaded: () => import_controlComponents.ClerkLoaded,
  ClerkLoading: () => import_controlComponents.ClerkLoading,
  ClerkProvider: () => ClerkProvider,
  CreateOrganization: () => import_uiComponents.CreateOrganization,
  GoogleOneTap: () => import_uiComponents.GoogleOneTap,
  OrganizationList: () => import_uiComponents.OrganizationList,
  OrganizationProfile: () => import_uiComponents.OrganizationProfile,
  OrganizationSwitcher: () => import_uiComponents.OrganizationSwitcher,
  PricingTable: () => import_uiComponents.PricingTable,
  RedirectToCreateOrganization: () => import_controlComponents.RedirectToCreateOrganization,
  RedirectToOrganizationProfile: () => import_controlComponents.RedirectToOrganizationProfile,
  RedirectToSignIn: () => import_controlComponents.RedirectToSignIn,
  RedirectToSignUp: () => import_controlComponents.RedirectToSignUp,
  RedirectToTasks: () => import_controlComponents.RedirectToTasks,
  RedirectToUserProfile: () => import_controlComponents.RedirectToUserProfile,
  Show: () => Show,
  SignIn: () => import_uiComponents.SignIn,
  SignInButton: () => import_uiComponents.SignInButton,
  SignInWithMetamaskButton: () => import_uiComponents.SignInWithMetamaskButton,
  SignOutButton: () => import_uiComponents.SignOutButton,
  SignUp: () => import_uiComponents.SignUp,
  SignUpButton: () => import_uiComponents.SignUpButton,
  TaskChooseOrganization: () => import_uiComponents.TaskChooseOrganization,
  TaskResetPassword: () => import_uiComponents.TaskResetPassword,
  TaskSetupMFA: () => import_uiComponents.TaskSetupMFA,
  UNSAFE_PortalProvider: () => import_controlComponents.UNSAFE_PortalProvider,
  UserAvatar: () => import_uiComponents.UserAvatar,
  UserButton: () => import_uiComponents.UserButton,
  UserProfile: () => import_uiComponents.UserProfile,
  Waitlist: () => import_uiComponents.Waitlist,
  getToken: () => import_getToken.getToken,
  useAuth: () => import_hooks.useAuth,
  useClerk: () => import_hooks.useClerk,
  useEmailLink: () => import_hooks.useEmailLink,
  useOrganization: () => import_hooks.useOrganization,
  useOrganizationCreationDefaults: () => import_hooks.useOrganizationCreationDefaults,
  useOrganizationList: () => import_hooks.useOrganizationList,
  useReverification: () => import_hooks.useReverification,
  useSession: () => import_hooks.useSession,
  useSessionList: () => import_hooks.useSessionList,
  useSignIn: () => import_hooks.useSignIn,
  useSignUp: () => import_hooks.useSignUp,
  useUser: () => import_hooks.useUser,
  useWaitlist: () => import_hooks.useWaitlist
});
module.exports = __toCommonJS(index_exports);
var import_controlComponents = require("./client-boundary/controlComponents");
var import_uiComponents = require("./client-boundary/uiComponents");
var import_hooks = require("./client-boundary/hooks");
var import_getToken = require("@clerk/shared/getToken");
var ComponentsModule = __toESM(require("#components"));
const ClerkProvider = ComponentsModule.ClerkProvider;
const Show = ComponentsModule.Show;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  APIKeys,
  AuthenticateWithRedirectCallback,
  ClerkDegraded,
  ClerkFailed,
  ClerkLoaded,
  ClerkLoading,
  ClerkProvider,
  CreateOrganization,
  GoogleOneTap,
  OrganizationList,
  OrganizationProfile,
  OrganizationSwitcher,
  PricingTable,
  RedirectToCreateOrganization,
  RedirectToOrganizationProfile,
  RedirectToSignIn,
  RedirectToSignUp,
  RedirectToTasks,
  RedirectToUserProfile,
  Show,
  SignIn,
  SignInButton,
  SignInWithMetamaskButton,
  SignOutButton,
  SignUp,
  SignUpButton,
  TaskChooseOrganization,
  TaskResetPassword,
  TaskSetupMFA,
  UNSAFE_PortalProvider,
  UserAvatar,
  UserButton,
  UserProfile,
  Waitlist,
  getToken,
  useAuth,
  useClerk,
  useEmailLink,
  useOrganization,
  useOrganizationCreationDefaults,
  useOrganizationList,
  useReverification,
  useSession,
  useSessionList,
  useSignIn,
  useSignUp,
  useUser,
  useWaitlist
});
