// @clerk/nextjs/dist/cjs/client-boundary/hooks.js
"use strict";
"use client";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var hooks_exports = {};
__export(hooks_exports, {
  EmailLinkErrorCode: () => import_errors.EmailLinkErrorCode,
  EmailLinkErrorCodeStatus: () => import_errors.EmailLinkErrorCodeStatus,
  isClerkAPIResponseError: () => import_errors.isClerkAPIResponseError,
  isClerkRuntimeError: () => import_errors.isClerkRuntimeError,
  isEmailLinkError: () => import_errors.isEmailLinkError,
  isKnownError: () => import_errors.isKnownError,
  isMetamaskError: () => import_errors.isMetamaskError,
  isReverificationCancelledError: () => import_errors.isReverificationCancelledError,
  useAuth: () => import_react.useAuth,
  useClerk: () => import_react.useClerk,
  useEmailLink: () => import_react.useEmailLink,
  useOrganization: () => import_react.useOrganization,
  useOrganizationCreationDefaults: () => import_react.useOrganizationCreationDefaults,
  useOrganizationList: () => import_react.useOrganizationList,
  useReverification: () => import_react.useReverification,
  useSession: () => import_react.useSession,
  useSessionList: () => import_react.useSessionList,
  useSignIn: () => import_react.useSignIn,
  useSignUp: () => import_react.useSignUp,
  useUser: () => import_react.useUser,
  useWaitlist: () => import_react.useWaitlist
});
module.exports = __toCommonJS(hooks_exports);
var import_react = require("@clerk/react");
var import_errors = require("@clerk/react/errors");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  EmailLinkErrorCode,
  EmailLinkErrorCodeStatus,
  isClerkAPIResponseError,
  isClerkRuntimeError,
  isEmailLinkError,
  isKnownError,
  isMetamaskError,
  isReverificationCancelledError,
  useAuth,
  useClerk,
  useEmailLink,
  useOrganization,
  useOrganizationCreationDefaults,
  useOrganizationList,
  useReverification,
  useSession,
  useSessionList,
  useSignIn,
  useSignUp,
  useUser,
  useWaitlist
});
