// @clerk/nextjs/dist/esm/server/proxy.js
import "../chunk-BUSYA2B4.js";
import {
  clerkFrontendApiProxy as backendProxy,
  DEFAULT_PROXY_PATH
} from "@clerk/backend/proxy";
import { PUBLISHABLE_KEY, SECRET_KEY } from "./constants";
import { DEFAULT_PROXY_PATH as DEFAULT_PROXY_PATH2 } from "@clerk/backend/proxy";
async function clerkFrontendApiProxy(request, options) {
  return backendProxy(request, {
    proxyPath: (options == null ? void 0 : options.proxyPath) || DEFAULT_PROXY_PATH,
    publishableKey: (options == null ? void 0 : options.publishableKey) || PUBLISHABLE_KEY,
    secretKey: (options == null ? void 0 : options.secretKey) || SECRET_KEY
  });
}
function createFrontendApiProxyHandlers(options) {
  const handler = async (request) => {
    return clerkFrontendApiProxy(request, options);
  };
  return {
    GET: handler,
    POST: handler,
    PUT: handler,
    DELETE: handler,
    PATCH: handler
  };
}
export {
  DEFAULT_PROXY_PATH2 as DEFAULT_PROXY_PATH,
  clerkFrontendApiProxy,
  createFrontendApiProxyHandlers
};
