// @clerk/nextjs/dist/cjs/app-router/server/ClerkProvider.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ClerkProvider_exports = {};
__export(ClerkProvider_exports, {
  ClerkProvider: () => ClerkProvider
});
module.exports = __toCommonJS(ClerkProvider_exports);
var import_react = __toESM(require("react"));
var import_buildClerkProps = require("../../server/buildClerkProps");
var import_mergeNextClerkPropsWithEnv = require("../../utils/mergeNextClerkPropsWithEnv");
var import_ClerkProvider = require("../client/ClerkProvider");
var import_DynamicClerkScripts = require("./DynamicClerkScripts");
var import_keyless_provider = require("./keyless-provider");
var import_utils = require("./utils");
const getDynamicClerkState = import_react.default.cache(async function getDynamicClerkState2() {
  const request = await (0, import_utils.buildRequestLike)();
  const data = (0, import_buildClerkProps.getDynamicAuthData)(request);
  return data;
});
async function ClerkProvider(props) {
  const { children, dynamic, ...rest } = props;
  const statePromiseOrValue = dynamic ? getDynamicClerkState() : void 0;
  const propsWithEnvs = (0, import_mergeNextClerkPropsWithEnv.mergeNextClerkPropsWithEnv)({
    ...rest,
    // Even though we always cast to InitialState here, this might still be a promise.
    // While not reflected in the public types, we do support this for React >= 19 for internal use.
    initialState: statePromiseOrValue
  });
  const { shouldRunAsKeyless, runningWithClaimedKeys } = await (0, import_keyless_provider.getKeylessStatus)(propsWithEnvs);
  const scriptsSlot = dynamic ? /* @__PURE__ */ import_react.default.createElement(import_react.Suspense, null, /* @__PURE__ */ import_react.default.createElement(
    import_DynamicClerkScripts.DynamicClerkScripts,
    {
      publishableKey: propsWithEnvs.publishableKey,
      __internal_clerkJSUrl: propsWithEnvs.__internal_clerkJSUrl,
      __internal_clerkJSVersion: propsWithEnvs.__internal_clerkJSVersion,
      __internal_clerkUIUrl: propsWithEnvs.__internal_clerkUIUrl,
      __internal_clerkUIVersion: propsWithEnvs.__internal_clerkUIVersion,
      domain: propsWithEnvs.domain,
      proxyUrl: propsWithEnvs.proxyUrl,
      prefetchUI: propsWithEnvs.prefetchUI
    }
  )) : void 0;
  if (shouldRunAsKeyless) {
    return /* @__PURE__ */ import_react.default.createElement(
      import_keyless_provider.KeylessProvider,
      {
        rest: propsWithEnvs,
        runningWithClaimedKeys,
        __internal_scriptsSlot: scriptsSlot
      },
      children
    );
  }
  return /* @__PURE__ */ import_react.default.createElement(
    import_ClerkProvider.ClientClerkProvider,
    {
      ...propsWithEnvs,
      __internal_scriptsSlot: scriptsSlot
    },
    children
  );
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ClerkProvider
});
